### PR TITLE
Test on Django 5.0

### PR DESCRIPTION
**Pull request by adamchainz**
Repeating the changes from #284 for the newest Django.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,21 +15,46 @@ jobs:
...
+        - '3.9'
+        - '3.10'
+        - '3.11'
+        django:
+        - '3.2'
+        - '4.0'
+        - '4.1'
+        - '4.2'
+        - '5.0'
+        - 'main'
```
Comment (adamchainz) [inline .github/workflows/ci.yml:30]: broke out into one-per-line to reduce future diff noise when adding/removing versions

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,21 +15,46 @@ jobs:
...
         exclude:
           - python-version: '3.7'
             django: '4.0'
```
Comment (adamchainz) [inline .github/workflows/ci.yml:33]: I suggest a follow-up PR to remove 3.7, since it’s EOL. Sound good?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -11,13 +11,46 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
-        django: ['3.2', '4.0', '4.1']
```
Comment (adamchainz) [inline .github/workflows/release.yml:15]: I realized this duplicated actions file wasn't updated in the past 😕. Maybe there’s a way to reduce the duplication more.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (jsocol) [reply .github/workflows/ci.yml:33]: Absolutely—I'll take care of that, thank you!

File: .github/workflows/release.yml (hunk 3 above)
Comment (jsocol) [reply .github/workflows/release.yml:15]: I looked for ways to move the matrix into the task but I couldn't find any. It's probably OK to do a significantly reduced set here, anyway

**Review by jsocol — APPROVED** [review]
Thank you very much!

**Comment by adamchainz** [comment]
reface_1610231641821_692

**Pull request merged**